db.py

```diff
@@ -190,6 +190,14 @@ class PublishAssignment(db.Model):
     )
 
 
+class SaveAssignment(db.Model):
```

Comment:
Relational database systems are there to store related pieces of data and ensure the whole data integrity through operations. 

By storing various data, including their relations, in large TEXT/BLOB/JSON columns, you're breaking the embedded integrity check system and make the use of such a database system irrelevant, as it is not capable of parsing the column content.

There should be no difference between an assignment before and after publishing, except it is shown to users. Assignments can be saved in a unique table then, whether they are still part of a draft or not.